controllers/asset.js

```diff
@@ -18,9 +24,84 @@ const sanitize = function(query_argument) {
     return decodeURIComponent(query_argument);
 };
 
+// Utility to create the next reference url for paging
+const get_next_url = (url, next) => {
+    let parsed_url = _url.parse(url, true);
+    parsed_url.query.start = next;
+    parsed_url.search = undefined;  // If search is set it prevails over the query, we must unset it
+    return parsed_url.format();
+};
+
+// Get extension from path
+const substr_ext = path => _path.extname(path).split('.').join('');
+
 module.exports = function(server, context) {
     const _workspace = require('../assetmanager/workspace')(context);
 
+    // WORKSPACES
+    server.get(workspaces_assets_regexp, function(req, res, next) {
+        let handler = new Handler(req, res, next);
+        let workspace_identifier = decodeURIComponent(req.params[0]);
+        let asset_ref = decodeURIComponent(req.params[1]);
+        _workspace.find(workspace_identifier)
+            .then(workspace => {
+                let options = {
+                    filterName: req.query.ref,
+                    maxResults: parseInt(req.query.maxResults, 10),
+                    start: parseInt(req.query.start, 10),
+                    q: req.query.q
+                };
+                return Asset.find_asset_by_ref(workspace, asset_ref, options)
+                    .then(asset => {
+                        let output = asset.output;
+                        if (output.items) {
+                            output.items = output.items.map(ass => {
+                                ass.url = _path.join("/contentbrowser/workspaces/", workspace.get_guid(), ass.meta.ref);
+                                return ass;
+                            });
+                        }
+                        if (asset.indexOfMoreResults) {
+                            output.nextLink = get_next_url(req.url, asset.indexOfMoreResults);
+                        }
+                        handler.sendJSON(output, 200, substr_ext(asset_ref));
```

Comment:
This should be the asset MIME type
```suggestion
                        handler.sendJSON(output, 200, asset.meta.type);
```